manager/state/raft/raft.go

```diff
@@ -630,6 +633,8 @@ func (n *Node) Run(ctx context.Context) error {
 					// cancelAll, or by its own check of signalledLeadership.
 					n.wait.cancelAll()
 				} else if !wasLeader && rd.SoftState.RaftState == raft.StateLeader {
+					// If this node was the leader in the previous iteration,
+					// set wasLeader to make sure all outstanding proposals are cancelled.
```

Comment:
This case is actually when the node becomes the leader.

`!wasLeader` - wasn't the leader before
`rd.SoftState.RaftState == raft.StateLeader` - is the leader now

Reply:
Makes sense. I'll correct it.

Reply:
The name might be confusing. I think renaming it to `isLeader` would be okay. The idea was that it's a long-lived state variable, not an instantaneous value.

I think the reason I named it that way was that if it as `isLeader`, this line would be `} else if !isLeader && rd.SoftState.RaftState == raft.StateLeader {`, which seems kind of nonsensical - we're the leader and we're not the leader. `wasLeader` tried to convey that it was the past state.